### PR TITLE
Change JRE in -version output

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Remove it from -Xinternalversion
- Change it to VERSION in -version

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

File: runtime/jcl/common/jclcinit.c [hunk 1]
```diff
@@ -142,7 +142,7 @@ jint computeFullVersionString(J9JavaVM* vm)
 #endif /* VENDOR_SHORT_NAME && VENDOR_SHA */
 
 	if (BUFFER_SIZE <= j9str_printf(PORTLIB, vminfo, BUFFER_SIZE + 1,
-			"JRE %s %s %s-%s %s" JIT_INFO J9VM_VERSION_STRING OMR_INFO VENDOR_INFO OPENJDK_INFO,
+			"RUNTIME %s %s %s-%s %s" JIT_INFO J9VM_VERSION_STRING OMR_INFO VENDOR_INFO OPENJDK_INFO,
```
Comment (JasonFengJ9) [inline runtime/jcl/common/jclcinit.c:145]: Is there any test checking the initial `JRE` occurrence? `semeru` product name change caused various failures.

File: runtime/vm/jvminit.c [hunk 2]
```diff
@@ -1699,11 +1699,11 @@ j9print_internal_version(J9PortLibrary *portLib) {
 
 #if defined(OPENJ9_BUILD)
 #if defined(J9JDK_EXT_VERSION) && defined(J9JDK_EXT_NAME)
-	j9tty_err_printf(PORTLIB, "Eclipse OpenJ9 %s %s-bit Server VM (%s) from %s-%s JRE with %s %s, built on %s %s by %s with %s\n",
+	j9tty_err_printf(PORTLIB, "Eclipse OpenJ9 %s %s-bit Server VM (%s) from %s-%s with %s %s, built on %s %s by %s with %s\n",
 		J9PRODUCT_NAME, J9TARGET_CPU_BITS, J9VERSION_STRING, J9TARGET_OS, J9TARGET_CPU_OSARCH,
 		J9JDK_EXT_NAME, J9JDK_EXT_VERSION,__DATE__, __TIME__, J9USERNAME, J9COMPILER_VERSION_STRING);
 #else
-        j9tty_err_printf(PORTLIB, "Eclipse OpenJ9 %s %s-bit Server VM (%s) from %s-%s JRE, built on %s %s by %s with %s\n",
+        j9tty_err_printf(PORTLIB, "Eclipse OpenJ9 %s %s-bit Server VM (%s) from %s-%s, built on %s %s by %s with %s\n",
```
Comment (JasonFengJ9) [inline runtime/vm/jvminit.c:1706]: Similar comment here.

**Comment by AdamBrousseau** [comment]
I think I've fixed known test cases except for the following
https://github.com/eclipse-openj9/openj9/blob/master/test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java#L620
```
	public static final String NATIVEMEMINFO_SUCCESS_KEYS = "JRE:,VM:,Classes:,Memory Manager,Java Heap:,Other:,Threads:,Java Stack:, bytes, allocations";
```

File: runtime/j9vm/j9memcategories.c [hunk 3]
```diff
@@ -37,8 +37,13 @@
  * - The first category is the root, all categories in the tree must be reachable from that.
  */
 
+#if defined(OPENJ9_BUILD)
+RUNTIME_INFO = "RUNTIME"
+#else
+RUNTIME_INFO = "JRE"
+#endif
```
Comment (JasonFengJ9) [inline runtime/j9vm/j9memcategories.c:44]: Suggest move these macros into a common place like `j9.h` to avoid code duplication.

**Comment by JasonFengJ9** [comment]
> public static final String NATIVEMEMINFO_SUCCESS_KEYS = "JRE:,VM:,Classes:,Memory Manager,Java Heap:,Other:,Threads:,Java Stack:, bytes, allocations";

I think a straightforward way is to remove `JRE:` from the success keys, a preferred way is to modify `DDRExtTesterBase.validate()` to accept a regex expression like `JRE| RUNTIME`.

File: runtime/jcl/common/jclcinit.c [hunk 4]
```diff
@@ -142,7 +142,7 @@ jint computeFullVersionString(J9JavaVM* vm)
 #endif /* VENDOR_SHORT_NAME && VENDOR_SHA */
 
 	if (BUFFER_SIZE <= j9str_printf(PORTLIB, vminfo, BUFFER_SIZE + 1,
-			"JRE %s %s %s-%s %s" JIT_INFO J9VM_VERSION_STRING OMR_INFO VENDOR_INFO OPENJDK_INFO,
+			RUNTIME_INFO " %s %s %s %s-%s %s" JIT_INFO J9VM_VERSION_STRING OMR_INFO VENDOR_INFO OPENJDK_INFO,
```
Comment (JasonFengJ9) [inline runtime/jcl/common/jclcinit.c:145]: An extra `%s` here.

File: runtime/vm/jvminit.c (hunk 2 above)
Comment (pshipton) [reply runtime/vm/jvminit.c:1706]: Jason pls clarify. "JRE" was removed here. What are you suggesting should be changed?

Comment (JasonFengJ9) [reply runtime/vm/jvminit.c:1713]: Similar comment referred https://github.com/eclipse-openj9/openj9/pull/13190#discussion_r669559136
> Is there any test checking the initial JRE occurrence? semeru product name change caused various failures.

Not about the change itself, if no test checking `JRE` literal here, the change is fine.
A personal build or PR build can find out.

Comment (pshipton) [reply runtime/vm/jvminit.c:1713]: Yes, I'll run some PR testing.

**Comment by pshipton** [comment]
jenkins compile jdk11 win

**Comment by JasonFengJ9** [comment]
`ERROR: Incorrect syntax for requested JDK version:'win'`

**Comment by AdamBrousseau** [comment]
jenkins compile win jdk11

**Comment by pshipton** [comment]
jenkins test sanity,extended zlinux jdk8

**Comment by pshipton** [comment]
win sanity https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/143
extended https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/144

**Comment by pshipton** [comment]
jenkins test sanity.system,extended.system zlinux jdk8

**Comment by pshipton** [comment]
zlinux system testing
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/147
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/148

File: test/functional/DDR_Test/src/j9vm/test/ddrext/DDRExtTesterBase.java [hunk 5]
```diff
@@ -531,8 +531,8 @@ protected int getJavaVersion() {
 			return javaVersion;
 		}
 		String[] coreInfo = getCoreInfo();
-		Pattern versionMatcher = Pattern.compile("JRE\\s+1\\.(\\d+).*"); /* look for JRE 1.n, where "n" is a string of digits */
-		Pattern versionMatcherForJava9 = Pattern.compile("JRE\\s(\\d+).*"); /* look for JRE 9 */
+		Pattern versionMatcher = Pattern.compile("JRE|RUNTIME\\s+1\\.(\\d+).*"); /* look for JRE 1.n, where "n" is a string of digits */
+		Pattern versionMatcherForJava9 = Pattern.compile("JRE|RUNTIME\\s(\\d+).*"); /* look for JRE 9 */
```
Comment (pshipton) [inline test/functional/DDR_Test/src/j9vm/test/ddrext/DDRExtTesterBase.java:535]: This isn't working.

**Comment by pshipton** [comment]
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/144/tapResults/

**Comment by pshipton** [comment]
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/144/tapResults/
testDDRExt_SharedClasses_0 failed, the changes to the regular expression in DDRExtTesterBase.java aren't working.

File: test/functional/DDR_Test/src/j9vm/test/ddrext/DDRExtTesterBase.java (hunk 5 above)
Comment (pshipton) [reply test/functional/DDR_Test/src/j9vm/test/ddrext/DDRExtTesterBase.java:535]: There should be brackets `(JRE|RUNTIME)`

**Comment by AdamBrousseau** [comment]
This is not required at the moment

**Pull request closed without merging**